### DISCUSSION TITLE
Setting Position, Gravity AND Rotation for multiple images

### DISCUSSION THREAD

**Discussion by styler2go** [ImageMagick/ImageMagick#3637] [Help] [answered]
Hello,

i am looking for help with my imagemagick script. I am trying to generate a 1330px width png with one star on the beginning and one star at the end. Both with 20px from the border. The last star should also be rotated on its own axis (in the middle).

How it should look:
![example](https://user-images.githubusercontent.com/2152144/117290118-b4a03a80-ae6d-11eb-8baf-0c5d76d8a506.png)

What my current command looks like:
```
convert -size "1330x259" xc:"transparent" -background "transparent" 
( "whitestar.png" -virtual-pixel transparent -repage "+20+0" -gravity "West" ) 
( "whitestar.png" -virtual-pixel transparent -repage "+20+0" -gravity "East" +distort SRT "45" ) 
-flatten rating.png
```

How the result currently looks like:
![rating](https://user-images.githubusercontent.com/2152144/117290727-6d667980-ae6e-11eb-9fe9-fe99c89569bf.png)

What works:
- Rotating in its own axis
- Drawing both stars

What doesn't work:
- Moving the star to the right by using gravity
- 20px distance from the left

I tried a lot of different things but i just couldn't figure out how to combine all of these options.

The star i am using: 
![whitestar](https://user-images.githubusercontent.com/2152144/117289584-0bf1db00-ae6d-11eb-8d4c-e01a84278a55.png)

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform? I will assume IM v7 on Windows BAT.

Many methods can be used. One is:
```
magick ^
  -size "1330x259" xc:"transparent" -background "transparent" ^
  ( "whitestar.png" -virtual-pixel transparent -repage "+20+0" -gravity "West" ) ^
  -composite ^
  ( "whitestar.png" -virtual-pixel transparent -repage "+20+0" -gravity "East" +distort SRT "45" ) ^
  -composite ^
  rating.png
```
EDIT to add: When using "-composite", "-repage +20+0" has no effect. You can use "-geometry +20+0" instead, or remove it.

**Reply by styler2go** [reply]
Awesome, this already does the trick. Thanks!
